src/main/java/com/nitnelave/CreeperHeal/block/CreeperContainer.java

```diff
@@ -0,0 +1,140 @@
+package com.nitnelave.CreeperHeal.block;
+
+import com.nitnelave.CreeperHeal.config.CfgVal;
+import com.nitnelave.CreeperHeal.config.CreeperConfig;
+import com.nitnelave.CreeperHeal.config.WCfgVal;
+import org.bukkit.Chunk;
+import org.bukkit.Location;
+import org.bukkit.World;
+import org.bukkit.block.BlockState;
+import org.bukkit.block.Container;
+import org.bukkit.inventory.DoubleChestInventory;
+import org.bukkit.inventory.Inventory;
+import org.bukkit.inventory.ItemStack;
+
+import java.util.Collection;
+import java.util.concurrent.ThreadLocalRandom;
+
+/**
+ * InventoryHolder implementation of CreeperMultiblock.
+ *
+ * @author Jikoo
+ */
+class CreeperContainer extends CreeperMultiblock
+{
+
+    /*
+     * Constructor.
+     */
+    CreeperContainer(Container blockState)
+    {
+        super(blockState);
+
+        Inventory inv = blockState.getInventory();
+        if (!(inv instanceof DoubleChestInventory))
+            return;
+
+        DoubleChestInventory doubleChest = ((DoubleChestInventory) inv);
+
+        BlockState right = doubleChest.getRightSide().getLocation().getBlock().getState();
+
+        // Left side is primary chest inventory
+        this.blockState = doubleChest.getLeftSide().getLocation().getBlock().getState();
+        this.dependents.add(right);
+
+    }
+
+    /*
+     * @see com.nitnelave.CreeperHeal.block.Replaceable#remove()
+     */
+    @Override
+    public void remove()
+    {
+        World world = blockState.getWorld();
+        if (CreeperConfig.getWorld(world).getBool(WCfgVal.DROP_CHEST_CONTENTS))
+        {
+            Location location = blockState.getLocation().add(0.5, 0.5, 0.5);
+            for (ItemStack itemStack : ((Container) blockState).getSnapshotInventory().getContents())
+                if (itemStack != null)
+                    world.dropItemNaturally(location, itemStack);
+            ((Container) blockState).getSnapshotInventory().clear();
+            blockState.update(true, false);
+            for (BlockState dependent : dependents)
+            {
+                if (!(dependent instanceof Container))
+                    continue;
+                location = dependent.getLocation().add(0.5, 0.5, 0.5);
```

Comment:
We should extract dropping the contents into a function, to avoid duplicate code.